### ISSUE TITLE
Relationship belongsTo with targetKey not working?

### ISSUE THREAD

**Issue by DanielTimLee** [sequelize/sequelize#7688]
_Thanks for wanting to report an issue you've found in Sequelize. Please delete this text and fill in the template below. If unsure about something, just do as best as you're able._

_Note that it will be much easier for us to fix the issue if a test case that reproduces the problem is provided. Ideally this test case should not have any external dependencies. We understand that it is not always possible to reduce your code to a small test case, but we would appreciate to have as much data as possible. Thank you!_

_Github should only be used for feature requests and bugs - all questions belong on StackOverflow, Slack, or Google groups._

_This template is for submitting issues - if you want to submit a feature request you may skip this template_

## What you are doing?
Relationship belongsTo with targetKey not working?
I followed this tutorial here, but I think it doesn't work properly.

http://docs.sequelizejs.com/manual/tutorial/associations.html#target-keys

```js
var User = sql.define('user', {/* attributes */});
var Company  = sql.define('company', {
  name:{
    type: SQL.STRING,
    allowNull:false
  }
});

User.belongsTo(Company, {foreignKey: 'fk_companyname', targetKey: 'name'}); 
// Adds fk_companyname to User
```

## What do you expect to happen?

Expected to Add `fk_companyname` to `users` (Automatic Plural table name) table.

## What is actually happening?
But the output was ER_CANNOT_ADD_FOREIGN!!

_Output, either JSON or SQL_
```
Unhandled rejection SequelizeBaseError: ER_CANNOT_ADD_FOREIGN: Cannot add foreign key constraint
    at Query.formatError (~/project/node_modules/sequelize/lib/dialects/mysql/query.js:175:14)
    at Query._callback (~/project/sequelize/lib/dialects/mysql/query.js:49:21)
    at Query.Sequence.end (~/project/node_modules/mysql/lib/protocol/sequences/Sequence.js:86:24)
    at Query.ErrorPacket (~/project/node_modules/mysql/lib/protocol/sequences/Query.js:88:8)
    at Protocol._parsePacket (~/project/node_modules/mysql/lib/protocol/Protocol.js:280:23)
    at Parser.write (~/project/node_modules/mysql/lib/protocol/Parser.js:75:12)
    at Protocol.write (~/project/node_modules/mysql/lib/protocol/Protocol.js:39:16)
    at Socket.<anonymous> (~/project/node_modules/mysql/lib/Connection.js:103:28)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:191:7)
    at readableAddChunk (_stream_readable.js:178:18)
    at Socket.Readable.push (_stream_readable.js:136:10)
    at TCP.onread (net.js:560:20)
```

__Dialect:__ **mysql** 
__Database version:__ 5.7.18 Homebrew (MacBook Pro (Retina, 13-inch, Early 2015) 10.12.4
__Sequelize version:__ 3.30.4

**Comment by harshithkashyap** [comment]
@DanielTimLee Kindly post the User model aswell. Have you verified that both foreignKey and targetKey use the same data type and that `name` field is a primary key?
You could be running into [this](https://stackoverflow.com/a/38638924) also. Ensure that `Model.sync` is called before creating associations.

**Comment by DanielTimLee** [comment]
@harshithkashyap  Thank you for Your comment!

Actually, I've just followed the tutorial (on Docs) , so those are the `Models` that I used.
(There are no any more attributes for the models.)

And also, I thought

Passing `targetKey` manually, It uses the column with `targetKey` even if that key column is `Primary key` or `not`.

(that's what I've understood with reading Docs)
(For now, the `name` column is not `Primary Key`)

I've Ensured `Model.sync` but it was not no use of ensuring it.
